src/ext/ddtrace.c

```diff
@@ -48,6 +50,130 @@
 #define PHP7_UNUSED(...) UNUSED(__VA_ARGS__)
 #endif
 
+/* {{{ arginfo */
+// DDTrace\Contracts\SpanContext
+ZEND_BEGIN_ARG_INFO_EX(arginfo_span_context_get_baggage_item, 0, 0, 1)
+ZEND_ARG_INFO(0, key)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO_EX(arginfo_span_context_with_baggage_item, 0, 0, 2)
+ZEND_ARG_INFO(0, key)
+ZEND_ARG_INFO(0, value)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO(arginfo_span_context_get_all_baggage_items, 0)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO(arginfo_span_context_get_propagated_priority_sampling, 0)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO_EX(arginfo_span_context_set_propagated_priority_sampling, 0, 0, 1)
+ZEND_ARG_INFO(0, propagatedPrioritySampling)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO(arginfo_span_context_is_host_root, 0)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO(arginfo_span_context_get_trace_id, 0)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO(arginfo_span_context_get_span_id, 0)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO(arginfo_span_context_get_parent_id, 0)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO(arginfo_span_context_is_dist_tracing_ctx, 0)
+ZEND_END_ARG_INFO()
+
+// DDTrace\Contracts\Tracer
+ZEND_BEGIN_ARG_INFO(arginfo_tracer_get_scope_manager, 0)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO(arginfo_tracer_get_active_span, 0)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO_EX(arginfo_tracer_start_active_span, 0, 0, 1)
+ZEND_ARG_INFO(0, operationName)
+ZEND_ARG_INFO(0, options)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO_EX(arginfo_tracer_start_integration_scope_and_span, 0, 0, 2)
+ZEND_ARG_OBJ_INFO(0, integration, DDTrace\\Integrations\\Integration, 0)
+ZEND_ARG_INFO(0, operationName)
+ZEND_ARG_INFO(0, options)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO_EX(arginfo_tracer_start_span, 0, 0, 1)
+ZEND_ARG_INFO(0, operationName)
+ZEND_ARG_INFO(0, options)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO_EX(arginfo_tracer_inject, 0, 0, 3)
+ZEND_ARG_OBJ_INFO(0, spanContext, DDTrace\\Contracts\\SpanContext, 0)
+ZEND_ARG_INFO(0, format)
+ZEND_ARG_INFO(1, carrier)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO_EX(arginfo_tracer_extract, 0, 0, 2)
+ZEND_ARG_INFO(0, format)
+ZEND_ARG_INFO(0, carrier)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO(arginfo_tracer_flush, 0)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO(arginfo_tracer_get_priority_sampling, 0)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO_EX(arginfo_tracer_start_root_span, 0, 0, 1)
+ZEND_ARG_INFO(0, operationName)
+ZEND_ARG_INFO(0, options)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO(arginfo_tracer_get_root_scope, 0)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO(arginfo_tracer_get_safe_root_scope, 0)
+ZEND_END_ARG_INFO()
+
+ZEND_BEGIN_ARG_INFO(arginfo_tracer_as_array, 0)
+ZEND_END_ARG_INFO()
+/* }}} */
+
+/* {{{ Function table */
+static const zend_function_entry php_ddtrace_span_context_funcs_interface[] = {
+    PHP_ABSTRACT_ME(SpanContext, getBaggageItem, arginfo_span_context_get_baggage_item) PHP_ABSTRACT_ME(
```

Comment:
Is the SpanContext used from within C code ? It looks to me like the `msgpack` serialization is quite generic and doesn't rely on it. Or did I miss something ?

My point is, is that if its not used from C level, then I wonder what os the reason for this functionality here

Reply:
I started to move all the interfaces down to the C level, but realized that I was probably a Bad Idea™. :) I think I'm going to revert these altogether.